packages/table/src/common/internal/scrollUtils.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { IRegion, RegionCardinality, Regions } from "../../regions";
+
+export function getScrollPositionForRegion(
+    region: IRegion,
+    currScrollLeft: number,
+    currScrollTop: number,
+    getLeftOffset: (columnIndex: number) => number,
+    getTopOffset: (rowIndex: number) => number,
+    numFrozenRows: number = 0,
+    numFrozenColumns: number = 0,
+) {
+    const cardinality = Regions.getRegionCardinality(region);
+
+    let nextTop = currScrollTop;
+    let nextLeft = currScrollLeft;
+
+    // if these were max-frozen-index values, we would have added 1 before passing to the get*Offset
+    // functions, but the counts are already 1-indexed, so we can just pass those.
+    const frozenColumnsCumulativeWidth = getLeftOffset(numFrozenColumns);
+    const frozenRowsCumulativeHeight = getTopOffset(numFrozenRows);
+
+    if (cardinality === RegionCardinality.CELLS) {
```

Comment:
for enums, i think we should use switch statements to make `default:` lintable.

Reply:
Ah, good thought. Will change.